tutorial/consuming_packages/intro_to_versioning.rst

```diff
@@ -0,0 +1,220 @@
+
+.. code-block:: python
+
+    def requirements(self):
+        self.requires("zlib/1.2.12#87a7211557b6690ef5bf7fc599dd8349")
+
+This mechanism can however be tedious to maintain and update when new revisions are created, so probably in the general case, this
+shouldn't be done.
+
+
+Lockfiles
+---------
+
+The usage of version ranges, and the possibility of creating new revisions of a given package without bumping the version allows
+to do automatic faster and more convenient updates, without need to edit recipes. 
+
+But in some occassions, there is also a need to provide an immutable and reproducible set of dependencies. This process is known
+as "locking", and the mechanism to allow it is "lockfile" files. A lockfile is a file that contains a fixed list of dependencies,
+specifying the exact version and exact revision. So, for example, a lockfile will never contain a version range with an expression,
+but only pinned dependencies. 
+
+A lockfile can be seen as a snapshot of a given dependency graph at some point in time.
+Such snapshot must be "realizable", that is, it needs to be a state that can be actually reproduce from the conanfile recipes.
+And this lockfile can be used at a later point in time to force that same state, even if there are new created package versions.
+
+Lets see lockfiles in action. First, lets pin the dependency to ``zlib/1.2.11`` in our example:
+
+
+.. code-block:: python
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+And lets capture a lockfile:
+
+.. code-block:: bash
+
+    conan lock create .
+
+    -------- Computing dependency graph ----------
+    Graph root
+        conanfile.py: .../conanfile.py
+    Requirements
+        zlib/1.2.11#4524fcdd41f33e8df88ece6e755a5dcc - Cache
+
+    Generated lockfile: .../conan.lock
+
+Lets see what the lockfile ``conan.lock`` contains:
+
+.. code-block:: json
+
+    {
+        "version": "0.5",
+        "requires": [
+            "zlib/1.2.11#4524fcdd41f33e8df88ece6e755a5dcc%1650538915.154"
+        ],
+        "build_requires": [],
+        "python_requires": []
+    }
+
+Now, lets restore the original ``requires`` version range:
+
+.. code-block:: python
+
+    def requirements(self):
+        self.requires("zlib/[~1.2]")
+
+
+And run ``conan install .``, which by default will find the ``conan.lock``, and run the equivalent ``conan install . --lockfile=conan.lock``
+
+.. code-block:: bash
+
+    conan install .
+
+    Graph root
+        conanfile.py: .../conanfile.py
+    Requirements
+        zlib/1.2.11#4524fcdd41f33e8df88ece6e755a5dcc - Cache
+
+
+Note how the version range is no longer resolved, and it doesn't get the ``zlib/1.2.12`` dependency, even if it is the 
+allowed range ``zlib/[~1.2]``, because the ``conan.lock`` lockfile is forcing it to stay in ``zlib/1.2.11`` and that exact revision too.
+
+
+Read more
+---------
+
+Go to :ref:`this section <tutorial_versioning_ci>` to learn more.
+
```

Comment:
.. container:: examples

    - :ref:`Introduction to Versioning and Continuous Integration <tutorial_versioning_ci>`